### PR TITLE
JBPM-7586 - SpringBoot - test coverage for KIE Server spring boot apps

### PR DESCRIPTION
**Pull request by MarianMacik**
Introducing SpringBoot test coverage.

So far WIP.

### CODE REVIEW

**Comment by MarianMacik** [comment]
jenkins retest this

File: kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+kieserver.casemgmt.enabled=${kieserver.casemgmt.enabled}
+kieserver.optaplanner.enabled=${kieserver.optaplanner.enabled}
+kieserver.swagger.enabled=${kieserver.swagger.enabled}
+
+# only required for jBPM
+#data source configuration
+spring.datasource.username=${org.kie.server.datasource.username}
+spring.datasource.password=${org.kie.server.datasource.password}
+spring.datasource.url=${org.kie.server.datasource.connection.url}
+spring.datasource.xa.data-source-class-name=${org.kie.server.datasource.driver.class}
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties]: @mswiderski Do you have more info about this? Is there some replacement?

[the remark this answers is not in the document]
Comment (mswiderski) [reply kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties]: there is no need for it at all, spring/narayana will figure it out based on the driver as far as I know

[the remark this answers is not in the document]
Comment (mswiderski) [reply kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties]: after additional thinking let's keep this in for now.

**Comment by mswiderski** [comment]
@sutaakar I made few adjustments and added case-authorisation.properties file the the integration-tests-sample - spring boot app. Have a look if all looks ok and if so I'll merge it

File: kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (sutaakar) [reply kie-server-parent/kie-server-tests/src/test/filtered-resources/application.properties]: FYI It seems that spring isn't able to figure out Sybase datasource class without this configuration, other databases seem ok.

File: kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-integ-tests-sample/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.kie</groupId>
+    <artifactId>kie-spring-boot-samples</artifactId>
+    <version>7.11.0-SNAPSHOT</version>
```
Comment (sutaakar) [inline kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-integ-tests-sample/pom.xml]: Please update the version to 7.12.0-SNAPSHOT

Comment (mswiderski) [reply kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-integ-tests-sample/pom.xml]: done

**Comment by mswiderski** [comment]
test failures not related

**Pull request merged**